common/Geometry2d/Rect.cpp

```diff
@@ -18,6 +18,111 @@ bool Rect::intersects(const Rect& other) const {
     }
 }
 
+int Rect::CohenSutherlandCode(const Point& other) const{
+    const int INSIDE = 0; // 0000
+    const int LEFT = 1;   // 0001
+    const int RIGHT = 2;  // 0010
+    const int BOTTOM = 4; // 0100
+    const int TOP = 8;    // 1000
+    int code;
+    double x,y;
+    x = other.x();
+    y = other.y();
+
+    code = INSIDE;          // initialised as being inside of [[clip window]]
+    if (x < minx())           // to the left of clip window
+        code |= LEFT;
+    else if (x > maxx())      // to the right of clip window
+        code |= RIGHT;
+    if (y < miny())           // below the clip window
+        code |= BOTTOM;
+    else if (y > maxy())      // above the clip window
+        code |= TOP;
+
+    return code;
+}
+
+bool Rect::intersects_(const Segment& other) const{
```

Comment:
You shouldn't need the underscore as C++ will allow method overloads depending on the arguments

Reply:
it was getting f***ed up in boost and I didn't really want to figure out how to fix it....